tests/integration/security/mtls/strict_test.go

```diff
@@ -43,12 +43,10 @@ func TestMtlsStrict(t *testing.T) {
 					Namespace:           systemNM,
 					RequiredEnvironment: environment.Kube,
 					Include: func(src echo.Instance, opts echo.CallOptions) bool {
-						// Exclude calls to the headless TCP port.
-						if opts.Target == rctx.Headless && opts.PortName == "tcp" {
-							return false
-						}
-
-						return true
+						// Exclude calls to the headless service.
+						// Auto mtls does not apply to headless service, because for headless service
+						// the cluster discovery type is ORIGINAL_DST, and it will not apply upstream tls setting
+						return opts.Target != rctx.Headless
```

Comment:
i understand the ORIGINAL_DST is a passthrough cluster, does not have TLS config. But does the corresponding headless listener has the tls config? Why? Could you show me the YAML of the headless svc/deployment yaml, and a config dump of the listener

Reply:
also, regardless automtls, (it's on by default on master now though), we can't support with authn policy + destination rule for this kind of case before either, right? so i feel it can be a misleading to mention auto mtls in this case.

Reply:
Absolutely, in this test, it enables tls in server side, and it expects tls connection. But for headless service, the discovery type is ORIGINAL_DST, the downstream will not initiate tls connection.

Reply:
I do not mean automtls lead to this error. I mean the autoMtls here does not take effect on headless service.